src/solver/optim/ortoolsImpl/include/antares/solver/optim/ortoolsImpl/linearProblem.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+
+#include <antares/solver/optim/api/linearProblem.h>
+#include <antares/solver/optim/ortoolsImpl/mipConstraint.h>
+#include <antares/solver/optim/ortoolsImpl/mipSolution.h>
+#include <antares/solver/optim/ortoolsImpl/mipVariable.h>
+
+// forward declaration
+namespace operations_research
+{
+class MPSolver;
+class MPSolverParameters;
+class MPObjective;
+} // namespace operations_research
+
+namespace Antares::Solver::Optim::OrtoolsImpl
+{
+
+class OrtoolsLinearProblem final: public Api::ILinearProblem
+{
+public:
+    OrtoolsLinearProblem(bool isMip, const std::string& solverName);
+    ~OrtoolsLinearProblem() final = default;
+
+    OrtoolsMipVariable* addNumVariable(double lb, double ub, const std::string& name) override;
+    OrtoolsMipVariable* addIntVariable(double lb, double ub, const std::string& name) override;
+    OrtoolsMipVariable* addVariable(double lb,
```

Comment:
this, in fact, was the original idea behind this method. but then we thought with @payetvin that there was no reason to hide it since it allows the user a little more flexibility (you can see that google made the choice to keep it public in MPSolver too). but if you think that it is confusing, we can hide it.

Reply:
As a suggestion, please see [this PR](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2385)